src/tournament_commands/player_commands/decklist.rs

```diff
@@ -5,5 +5,35 @@ use serenity::prelude::*;
 #[command("decklist")]
 #[description("Prints out one of your decklists.")]
 async fn decklist(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
-    todo!()
+     /* Get references to needed data from context */
+     let data = ctx.data.read().await;
+     let name_and_id = data.get::<TournamentNameAndIDMapContainer>().unwrap();
+     let gld_tourns = data.get::<GuildAndTournamentIDMapContainer>().unwrap();
+     let player_name = msg.author.id.0.to_string();
+
+     let id_iter = gld_tourns.get_left_iter(&msg.guild_id.unwrap()).unwrap();
```

Comment:
Here, we need to turn the regular `Iter` behind `id_iter` into a `FilterIter`. We can tell it to only pass through certain items.

```suggestion
     let id_iter = gld_tourns.get_left_iter(&msg.guild_id.unwrap()).unwrap().filter(|id| tourn.get(id).unwrap().players.contains(msg.user.id));
```